tests/regression/test_projection.py

```diff
@@ -246,29 +246,35 @@ def test_projector_expression():
 @pytest.mark.parametrize('tensor', ['scalar', 'vector', 'tensor'])
 def test_projector_bcs(tensor):
     mesh = UnitSquareMesh(2, 2)
+    x = SpatialCoordinate(mesh)
     if tensor == 'scalar':
         V = FunctionSpace(mesh, "CG", 1)
         V_ho = FunctionSpace(mesh, "CG", 5)
         bcs = [DirichletBC(V_ho, Constant(0.5), (1, 3)),
                DirichletBC(V_ho, Constant(-0.5), (2, 4))]
-        expr = Expression('cos(x[0]*pi*2)*sin(x[1]*pi*2)')
+        fct = cos(x[0]*pi*2)*sin(x[1]*pi*2)
+        v = Function(V).interpolate(fct)
     elif tensor == 'vector':
         V = VectorFunctionSpace(mesh, "CG", 1)
         V_ho = VectorFunctionSpace(mesh, "CG", 5)
-        bcs = [DirichletBC(V_ho, Expression(("0.5", "0.5")), (1, 3)),
-               DirichletBC(V_ho, Expression(("-0.5", "-0.5")), (2, 4))]
-        expr = Expression(['cos(x[0]*pi*2)*sin(x[1]*pi*2)']*2)
+        bcs = [DirichletBC(V_ho, Constant((0.5, 0.5)), (1, 3)),
```

Comment:
If I just use tuples, I get a `ValueError` telling me that I have not provided "a valid DirichletBC expression", which is why I use `Constant`/UFL expressions. Can you give me an example of what you exactly mean? I may just not be using the right input.